tests/Core/Layer/Collector/TagValueRegexCollectorTest.php

```diff
@@ -0,0 +1,114 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Tests\Qossmic\Deptrac\Core\Layer\Collector;
+
+use PHPUnit\Framework\TestCase;
+use Qossmic\Deptrac\Contract\Config\Collector\TagValueRegexConfig;
+use Qossmic\Deptrac\Contract\Layer\InvalidCollectorDefinitionException;
+use Qossmic\Deptrac\Core\Ast\AstMap\ClassLike\ClassLikeReference;
+use Qossmic\Deptrac\Core\Ast\AstMap\ClassLike\ClassLikeToken;
+use Qossmic\Deptrac\Core\Ast\AstMap\ClassLike\ClassLikeType;
+use Qossmic\Deptrac\Core\Layer\Collector\TagValueRegexCollector;
+
+final class TagValueRegexCollectorTest extends TestCase
+{
+    private TagValueRegexCollector $collector;
+
+    protected function setUp(): void
+    {
+        parent::setUp();
+
+        $this->collector = new TagValueRegexCollector();
+    }
+
+    public static function dataProviderSatisfy(): iterable
+    {
+        yield [TagValueRegexConfig::create('@foo'), ['@foo' => ['']]];
+
+        yield [TagValueRegexConfig::create('@foo'), ['@foo' => ['anything']]];
+
+        yield [
+            TagValueRegexConfig::create('@foo-bar', '/some/'),
+            ['@xyz' => [''], '@foo-bar' => ['anything', 'something']],
+        ];
+
+        yield [
+            TagValueRegexConfig::create('@foo-bar')->match('!thing$!'),
+            ['@xyz' => [''], '@foo-bar' => ['anything', 'something']],
+        ];
+    }
+
+    /**
+     * @dataProvider dataProviderSatisfy
+     */
+    public function testSatisfy(TagValueRegexConfig $configuration, array $tags): void
+    {
+        $actual = $this->collector->satisfy(
+            $configuration->toArray(),
+            new ClassLikeReference(ClassLikeToken::fromFQCN('Dummy'), ClassLikeType::TYPE_CLASS, [], [], $tags)
+        );
+
+        self::assertTrue($actual);
+    }
+
+    public static function dataProviderNotSatisfy(): iterable
+    {
+        yield [TagValueRegexConfig::create('@foo'), ['@bar' => ['anything']]];
```

Comment:
Being nitpicky: 
The other provider also set a key for the yield, it would be nice to have them here to.

Reply:
done